### ISSUE TITLE
Default hosting of visited hyperdrives. 

### ISSUE THREAD

**Issue by eHorlyck** [beakerbrowser/beaker#1907]
I would suggest an option to set the "host this hyper drive" to default on for new sites.

I would like to turn hosting on such that all new Hyper drives I visit,  automatically gets hosted by me as well. I would be nice to then be able  to set at limit such as the latest xx number of drives and max xx Gb storage. 

- [ ]  ability to set "always host visited hyperdrives" on/off
       - [ ] ability to limit number of hosted sites
       - [ ] ability to limit storage amount used
       
Super excited to see this project grow, good job.

**Comment by eHorlyck** [comment]
These setting could maybe even be set in the initial install to strengthen the entire network.

**Comment by Bluebie** [comment]
This would create a very unexpected user experience, hosting a website means downloading the entire website to your computer, even if you just visited one page. This could easily blow through someone's cellular data quota. I think it's important that users go through an informed consent flow before hosting content, including the size of the content they'll be downloading.

It's also worth considering if this strengthens or harms the network. If behaviour like this trashed people's home internet experience by downloading large blocks of mostly unneeded data, people may get fed up and chuck Beaker in the trash, which would effectively be a kind of "leeching" in the network because users like that aren't likely to "seed" the hyperdrive for long.

**Comment by eHorlyck** [comment]
I agree with what you say. I may have expressed my suggestion wrong. I desire to flip a switch to host any hyper site I visit, without having to opt-in per site. I would then like to set a limit though. 

In my case, I have symmetric unlimited gigabit connection from an always-on machine. So if I could easily contribute to the network as a host I would.

**Comment by zootella** [comment]
I agree this default could really help Beaker sites and the network as a whole, and hope Beaker adds it.

There are a lot of different ways to keep the user in control, like:
* Showing the user the option during setup (having it on by default, but allowing the user to change that)
* Only automatically hosting small drives
* Only hosting drives for a certain amount of time after each last visit, like a month

**Comment by da2x** [comment]
See also issue #1664.
